verify/module.py

```diff
@@ -89,8 +95,79 @@ class Verify(commands.Cog):
         name="reverify", description="Reverification commands.", parent=verification
     )
 
-    def __init__(self, bot):
+    def __init__(self, bot: Strawberry):
         self.bot: Strawberry = bot
+        self.bounce_check.start()
+
+    def cog_unload(self):
+        self.bounce_check.cancel()
+
+    # Tasks
+
+    @tasks.loop(seconds=60.0)
+    async def bounce_check(self):
+        """Periodically connect to IMAP server and check for undelivered mail.
+        If such e-mails are found, they will be logged and user will be contacted.
+        """
+        # TODO Use embeds
+        unread_messages = self._check_inbox_for_errors()
+        for message in unread_messages:
+            guild: discord.Guild = self.bot.get_guild(int(message["guild"]))
+            user: discord.Member = self.bot.get_user(int(message["user"]))
+            channel: discord.TextChannel = guild.get_channel(int(message["channel"]))
+
+            if guild is None or user is None:
+                continue
+
+            db_users: list[VerifyMember] = VerifyMember.get(
+                guild_id=guild.id, user_id=user.id, status=VerifyStatus.PENDING
+            )
+            db_user = db_users[0] if len(db_users) == 1 else None
+
+            if not db_user:
+                continue
+
+            utx = i18n.TranslationContext(guild_id=guild.id, user_id=user.id)
+            await guild_log.warning(
+                user,
+                channel if channel else guild.text_channels[0],
+                "Could not deliver verification code: "
+                f"{message['subject']} (User ID {message['user']})",
+            )
+
+            error_msg: str = (
+                _(
+                    utx,
+                    (
+                        "I could not send the verification code, you've probably made "
+                        "a typo{address_part}. Invoke the command `/strip` "
+                        "before requesting a new code."
+                    ),
+                )
+                + "\n\n"
+                + _(
+                    utx,
+                    "If I'm wrong and the e-mail is correct, contact the moderator team.",
+                )
+            )
+            if (
+                not await utils.discord.send_dm(
+                    user,
+                    error_msg.format(address_part=f": `{db_user.address}`"),
+                )
+                and channel
+            ):
+                await channel.send(
```

Comment:
Not sure, but this looks like the error message will be visible for everyone in the channel. It should be possible to send the error message in the channel to the user, but only the user will be able to see the message, like private message in the channel.

Reply:
That unfortunately does not work if you don't have interaction with the suer (e.g. user running a command). It's a Discord inbuilt limitation.

Reply:
So yes, the message will be unfortunately public.